terraform/modules/ecr/main.tf

```diff
@@ -21,3 +21,25 @@ resource "aws_ecr_repository" "this" {
     scan_on_push = true
   }
 }
+
+data "aws_iam_policy_document" "this" {
+  statement {
+    effect = "Allow"
+
+    principals {
+      type        = "AWS"
+      identifiers = formatlist("arn:aws:iam::%s:*", var.accessible_from_accounts)

```

Comment:
Allows access for any user/role from these accounts. This may well be more than we want - restricting to the task execution role would cause some interesting headaches around ordering that it's not worth dealing with right now. (Apparently you can only apply one repository policy, so there's no nice getting around it by defining it with the role...)

Reply:
Scratch that - AWS gives an annoyingly unhelpful error that goes away when you remove wildcards from this, so going with specifying the role arns.